lib/artoo/adaptors/io/digital_pin.rb

```diff
@@ -50,14 +50,17 @@ def set_mode(mode)
           @mode = mode
 
           if mode == 'w'
-            File.open("#{ GPIO_PATH }/gpio#{ pin_num }/direction", "w") { |f| f.write(GPIO_DIRECTION_WRITE) }
-            @pin_file = File.open("#{ GPIO_PATH }/gpio#{ pin_num }/value", "w")
+            set_pin('w')
           elsif mode =='r'
-            File.open("#{ GPIO_PATH }/gpio#{ pin_num }/direction", "w") { |f| f.write(GPIO_DIRECTION_READ) }
-            @pin_file = File.open("#{ GPIO_PATH }/gpio#{pin_num}/value", "r")
+            set_pin('r')
           end
         end
 
+        def set_pin(mode)
+          File.open("#{ GPIO_PATH }/gpio#{ pin_num }/direction", "w") { |f| f.write(GPIO_DIRECTION_WRITE) }
```

Comment:
Shouldn't the data being written here be either `GPIO_DIRECTION_WRITE` or `GPIO_DIRECTION_READ` depending on direction?